modin/core/dataframe/pandas/partitioning/partition_manager.py

```diff
@@ -67,12 +67,9 @@ def wait(*args, **kwargs):
             # serially kick off all the deferred computations so that they can
             # all run asynchronously, then wait on all the results.
             [part.drain_call_queue() for part in partitions.flatten()]
-            # need to go through all the values of the map iterator
-            # since `wait` does not return anything, we need to explicitly add
-            # the return `True` value from the lambda
-            # TODO(https://github.com/modin-project/modin/issues/4491): Wait
-            # for all the partitions in parallel.
-            all(map(lambda partition: partition.wait() or True, partitions.flatten()))
+            # The partition manager invokes the relevant .wait() method under
+            # the hood, which should wait in parallel for all computations to finish
+            PandasDataframePartitionManager.wait_partitions(partitions.flatten())
```

Comment:
this calls `.wait_partitions()` on explicitly `PandasDataframePartitionManager`, not the manager for giving partitions (e.g. `PandasOnDaskDataframePartitionManager`), this seems to be wrong

Reply:
We should probably extract the partition manager out of `*args`.

```python
partition_mgr_cls = args[0]
```

Reply:
On further examination, it seems like this has to be `args[0]` rather than adding a `cls` argument to the decorator function (tests crash otherwise).

Reply:
got a link to tests' failure? looks very suspicious

Reply:
[Here](https://github.com/modin-project/modin/runs/7307878853?check_suite_focus=true), I was able to reproduce it locally as well. I can't find the portion in the official Python docs describing why `*args` and `**kwargs` are the only allowable options, but all the examples I've seen through searching seem to only use those.

Reply:
Can I see the link to the commit? I mean, maybe you forgot to pass this `cls` as the argument? (like I did in my suggestion)

Reply:
https://github.com/modin-project/modin/pull/4656/commits/4f91ce8bf9557efd2384fec30059883537c7533f

It seems you're right, I missed `func(cls, *args, **kwargs)`. I've put the explicit `cls` argument back.